core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -327,8 +326,14 @@ object SparkSubmit {
       printStream.println("\n")
     }
 
-    val loader = new ExecutorURLClassLoader(new Array[URL](0),
-      Thread.currentThread.getContextClassLoader)
+    val loader =
+      if (sysProps.getOrElse("spark.driver.userClassPathFirst", "false").toBoolean) {
+        new ChildFirstURLClassLoader(new Array[URL](0),
```

Comment:
That kinda clashes with Sandy's comment, which I agree with, that calling them "Executor" is misleading since they're used in other contexts. And frankly, compared to the other changes here, changing those class names is a very small one.